integration-tests/src/test/java/no/sikt/nva/nvi/events/DynamoDbUpdateEventTest.java

```diff
@@ -0,0 +1,367 @@
+  private DataEntryUpdateHandlerContext dataEntryUpdateHandlerContext;
+
+  @BeforeEach
+  void init() {
+    sharedQueueClient = new FakeSqsClient();
+    snsClient = new FakeNotificationClient();
+    dynamoDbEventHandlerContext = new DynamoDbToEventQueueHandlerContext(sharedQueueClient);
+    dataEntryUpdateHandlerContext = new DataEntryUpdateHandlerContext(sharedQueueClient, snsClient);
+  }
+
+  @Test
+  void shouldSendMessageBatchWithSize10() {
+    var dynamoDbEvent = randomEventWithNumberOfDynamoRecords(15);
+
+    dynamoDbEventHandlerContext.handleEvent(dynamoDbEvent);
+
+    var batchOneMessages = extractBatchEntryMessageBodiesAtIndex(0);
+    var batchTwoMessages = extractBatchEntryMessageBodiesAtIndex(1);
+    assertEquals(10, batchOneMessages.size());
+    assertEquals(5, batchTwoMessages.size());
+  }
+
+  @ParameterizedTest
+  @MethodSource("candidateEventProvider")
+  void shouldPassCandidateEventsToCorrectSnsTopic(
+      DynamodbEvent dynamoDbEvent, DynamoDbChangeMessage expectedMessage, String expectedTopic) {
+    processDynamoEvent(dynamoDbEvent);
+    var publishedMessages = snsClient.getPublishedMessages();
+    assertThat(publishedMessages)
+        .hasSize(1)
+        .allMatch(hasMessageBody(expectedMessage))
+        .allMatch(hasTopic(expectedTopic));
+  }
+
+  @ParameterizedTest
+  @MethodSource("approvalEventProvider")
+  void shouldPassApprovalEventsToCorrectSnsTopic(
+      DynamodbEvent dynamoDbEvent, DynamoDbChangeMessage expectedMessage, String expectedTopic) {
+    processDynamoEvent(dynamoDbEvent);
+    var publishedMessages = snsClient.getPublishedMessages();
+    assertThat(publishedMessages)
+        .hasSize(1)
+        .allMatch(hasMessageBody(expectedMessage))
+        .allMatch(hasTopic(expectedTopic));
+  }
+
+  @ParameterizedTest()
+  @MethodSource("otherDaoInsertEventProvider")
+  void shouldDoNothingWhenReceivingEventWithOtherDaoTypes(DynamodbEvent dynamoDbEvent) {
+    processDynamoEvent(dynamoDbEvent);
+
+    assertThat(snsClient.getPublishedMessages()).isEmpty();
+    assertThat(getDlqMessages()).isEmpty();
+  }
+
+  @ParameterizedTest
+  @EnumSource(
+      value = OperationType.class,
+      names = {"INSERT", "MODIFY", "REMOVE"})
+  void shouldHandleAllValidOperationTypes(OperationType operationType) {
+    var dynamoDbEvent = createValidCandidateEvent(operationType);
+    processDynamoEvent(dynamoDbEvent);
+    assertThat(getDlqMessages()).isEmpty();
+    assertThat(snsClient.getPublishedMessages()).hasSize(1);
+  }
+
+  @Test
+  void shouldSendMessageToDlqIfSendingBatchFails() {
+    var candidateIdentifier = randomUUID();
+    var dynamoDbEvent = createCandidateEvent(candidateIdentifier, OperationType.MODIFY, true);
+
+    // Rig the sharedQueueClient to throw an exception when sending a message to the queue
+    sharedQueueClient = spy(FakeSqsClient.class);
+    doThrow(SqsException.class).when(sharedQueueClient).sendMessageBatch(any(), any());
```

Comment:
Since FakeSqsClient is "our" code, should we not be able to throw when dispatching without this mockery?